code/obj/item/stun_baton.dm

```diff
@@ -557,14 +602,16 @@
 	update_icon()
 		if (!src || !istype(src))
 			return
-		switch (state)
+		switch (src.state)
 			if (CLOSED_AND_OFF)
-				set_icon_state(src.icon_off)
+				src.set_icon_state(src.icon_off)
+				src.item_state = src.item_off
 			if (OPEN_AND_ON)
-				set_icon_state(src.icon_on)
+				src.set_icon_state(src.icon_on)
+				src.item_state = src.item_on
 			if (OPEN_AND_OFF)
-				set_icon_state(src.icon_off_open)
-
+				src.set_icon_state(src.icon_off_open)
+				src.item_state = src.item_off_open
```

Comment:
Possibly out of the scope of this PR but would it maybe make sense for the `item_state` setting to happen nside `set_icon_state` or is there something that relies on them being different?

Reply:
maybe for when i eventually just entirely refactor batons. but thats in the future. not important right now. id rather have a seperate set_item_state proc. the item states and icon states dont always have the same name, and idk about forcing that?